app/controllers/static_controller.rb

```diff
@@ -27,9 +27,40 @@ class StaticController < ApplicationController
   }
   CUSTOM_PAGES = {} # Add via `#add_topic_static_page` in plugin API
 
+  def validate_redirect_param
+    redirect_path = params[:redirect]
+    if redirect_path.present?
+      raise Discourse::InvalidParameters.new(:redirect) unless redirect_path.is_a?(String)
+
+      begin
+        forum_uri = URI(Discourse.base_url)
```

Comment:
Nitpick: since we only check for the `host`, how about doing this instead?

```suggestion
        forum_host = URI(Discourse.base_url).host
```

and then do the following check?

```ruby
uri.host == forum_host
```